### PR TITLE
[hotfix][azure] fix ArchUnit flag

### PR DESCRIPTION
**Pull request by Airblader**
## What is the purpose of the change

This mistake actually prevented the CI from failing when violations have been fixed.

I expect this to fail the CI the first time as there are fixed violations that haven't been removed.
*Edit:* Yep, works now, [see here](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=30246&view=logs&j=5c8e7682-d68f-54d1-16a2-a09310218a49&t=86f654fa-ab48-5c1a-25f4-7e7f6afb9bba).

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Airblader** [comment]
@flinkbot run azure

File: azure-pipelines.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ stages:
             vmImage: 'ubuntu-20.04'
           e2e_pool_definition:
             vmImage: 'ubuntu-20.04'
-          environment: PROFILE="-Dhadoop.version=2.8.5 -Dinclude_hadoop_aws -Dscala-2.12 -Darchunit.freeze.default.allowStoreUpdate=false"
+          environment: PROFILE="-Dhadoop.version=2.8.5 -Dinclude_hadoop_aws -Dscala-2.12 -Darchunit.freeze.store.default.allowStoreUpdate=false"
```
Comment (zentol) [inline azure-pipelines.yml:80]: Since we want this to be applied for all tests you could add this to `test_controller.sh#MVN_TEST_OPTIONS`.

Comment (Airblader) [reply azure-pipelines.yml:80]: Ah, thanks. I'll move it there and re-add the violation entry to let CI confirm again that it does indeed still work.

**Review by zentol — APPROVED** [review]

**Comment by Airblader** [comment]
CI failed correctly again, pushed the removal of the violation and will merge once CI passes.

**Pull request merged**